File: packages/motif-vite/src/core/material/shell.ts
import type { Declaration, ParsedClass } from '../shared.js';
import { resolveMaterialShellFeedback } from './shell/feedback.js';
import { resolveMaterialShellLayout } from './shell/layout.js';

export function resolveMaterialShell(parsed: ParsedClass): Declaration[] | null {
  return resolveMaterialShellFeedback(parsed) ?? resolveMaterialShellLayout(parsed);
}

File: core/src/rule/fluent/shell.rs
#[path = "shell/feedback.rs"]
mod feedback;
#[path = "shell/layout.rs"]
mod layout;

use super::super::Declaration;
use crate::parse::ParsedClass;
use crate::token::TokenRegistry;

pub(super) fn resolve(parsed: &ParsedClass, tokens: &TokenRegistry) -> Option<Vec<Declaration>> {
    feedback::resolve(parsed, tokens).or_else(|| layout::resolve(parsed, tokens))
}
